<launch>
  <!-- Start your ROS 2 node -->
  <arg name="steer_can_id"  default="403105078"/> <!--1806E536 120-->
  <arg name="kelly_left_id"  default="217128454"/> <!--0CF11E06 100-->
  <arg name="kelly_right_id"  default="217128453"/> <!--0CF11E05 190 -->
  <arg name="gear_can_id"  default="217128709"/> <!--0CF11F05 191-->
  <arg name="can_send_id" default="105"/> <!--1806D9D9 403039337-->
  <arg name="base_frame_id"  default="base_link"/>
  <arg name="max_steer" default="735"/>
  <arg name="min_steer" default="435" />
  <arg name="left_steer" default="0.5" /> <!--in rad-->
  <arg name="right_steer" default="-0.5" /> <!--in rad-->
  <arg name="tire_radius" default="0.334"/>  <!--in m-->
  <arg name="wheel_base" default="2.75"/>  <!--in m-->
  <arg name="origin_lat" default="12.923903488321232"/> <!-- Origin latitude -->
  <arg name="origin_lon" default="77.50052742264235"/>  <!-- Origin longitude -->
  <arg name="origin_alt" default="0.0"/>

  
  <node pkg="wipod_interface" exec="wipod_interface" name="wipod_interface"  output="screen">
    <param name="steer_can_id"  value="$(var steer_can_id)"/>
    <param name="kelly_left_id" value="$(var kelly_left_id)"/>
    <param name="gear_can_id" value="$(var gear_can_id)"/>
    <param name="base_frame_id"  value="$(var base_frame_id)"/>
    <param name="can_send_id" value="$(var can_send_id)"/>
    <param name="max_steer"  value="$(var max_steer)"/>
    <param name="min_steer"  value="$(var min_steer)"/>
    <param name="left_steer"  value="$(var left_steer)"/>
    <param name="right_steer"  value="$(var right_steer)"/>
    <param name="tire_radius" value="$(var tire_radius)"/>
    <param name="wheel_base" value="$(var wheel_base)"/>
    <param name="origin_lat" value="$(var origin_lat)"/>
    <param name="origin_lon" value="$(var origin_lon)"/>
    <param name="origin_alt" value="$(var origin_alt)"/>
  </node>
</launch>